lib/typescript/httpclient/client.ts

```diff
@@ -59,15 +59,47 @@ interface EndpointDefinition<T, K = void> {
 }
 
 export class HttpClient {
-  public readonly apiUrlConfig?: string;
-  private readonly unauthorizedErrorCallback?: (body: any) => void;
+  public readonly apiUrl?: string;
+  public readonly loginUrl?: string;
+  private readonly unauthorizedErrorCallback?: (body: any, loginUrl: string) => void;
 
-  constructor(apiUrlConfig: string, unauthorizedErrorCallback?: (body: any) => void) {
-    this.apiUrlConfig = apiUrlConfig;
+  constructor(apiUrl: string, unauthorizedErrorCallback?: (body: any, loginUrl: string) => void) {
+    this.apiUrl = apiUrl;
+    this.loginUrl = `${apiUrl}/login`;
     this.unauthorizedErrorCallback = unauthorizedErrorCallback;
   }
 
+  private async doFetchFromBackend(url: string, body?: any): Promise<any> {
+    const headers = {
+      Accept: 'application/json',
+      'X-Requested-With': 'XMLHttpRequest',
+    };
+
+    if (!(body instanceof FormData)) {
+      if (!isString(body)) {
+        body = JSON.stringify(body);
+      }
+      headers['Content-Type'] = 'application/json';
+    }
+
+    const response: Response = await fetch(`${this.apiUrl}/${url}`, {
+      method: 'POST',
+      headers: headers,
+      mode: 'cors',
+      credentials: 'include',
+      body: body as BodyInit,
+    });
+
+    return this.parseBody(response);
+  }
+
   private async parseBody(response: Response): Promise<any> {
+    if (this.isAuthRedirect(response)) {
```

Comment:
This is not an optimal solution. I could not get Spring to return a `403` with the OAuth middleware. So instead we try and detect when we are being redirected and treat that as an authorization error.